cirq-core/cirq/linalg/transformations.py

```diff
@@ -595,7 +595,9 @@ def factor_state_vector(
     if validate:
         t2 = state_vector_kronecker_product(extracted, remainder)
         if not predicates.allclose_up_to_global_phase(t2, t1, atol=atol):
-            raise ValueError('The tensor cannot be factored by the requested axes')
+            if not np.isclose(np.linalg.norm(t1), 1):
+                raise ValueError('Input state must be normalized.')
+            raise EntangledStateError('The tensor cannot be factored by the requested axes')
```

Comment:
This is covered by the partial trace fallback tests, but we really should have an isolated test of the `EntangledStateError` path for `factor_state_vector`.